gms/gossiper.cc

```diff
@@ -1428,7 +1428,10 @@ endpoint_state& gossiper::get_or_create_endpoint_state(inet_address ep) {
     auto it = _endpoint_state_map.find(ep);
     if (it == _endpoint_state_map.end()) {
         auto eps = endpoint_state();
-        eps.add_application_state(application_state::RPC_ADDRESS, versioned_value::rpcaddress(ep));
+        if (utils::fb_utilities::is_me(ep)) {
+            auto rpc_addr = utils::fb_utilities::get_broadcast_rpc_address();
+            eps.add_application_state(application_state::RPC_ADDRESS, versioned_value::rpcaddress(rpc_addr));
```

Comment:
Why do we set RPC_ADDRESS here? Can we create an empty endpoint_state?

Reply:
It seems arbitrary that we initialize just this state here. There is much more states, they're added in storage_service.

Some states have dependencies between them, and they should appear together. It would be incorrect to add them at different points in time, and allow other nodes to observe inconsistent set of states. So why not leave adding all states to where it is done now, in storage service?

Reply:
I'm OK with making this change but what inter-dependencies the RPC_ADDRESS state have?